lib/ansible/plugins/action/reboot.py

```diff
@@ -187,69 +242,86 @@ def do_until_success_or_timeout(self, action, reboot_timeout, action_desc):
         except AnsibleConnectionFailure as e:
             # If the connection is closed too quickly due to the system being shutdown, carry on
-            display.debug('%s: AnsibleConnectionFailure caught and handled: %s' % (self._task.action, to_native(e)))
+            display.debug('{action}: AnsibleConnectionFailure caught and handled: {error}'.format(action=self._task.action, error=to_native(e)))
             reboot_result['rc'] = 0
 
         result['start'] = datetime.utcnow()
 
         if reboot_result['rc'] != 0:
             result['failed'] = True
             result['rebooted'] = False
-            result['msg'] = "Shutdown command failed. Error was %s, %s" % (
-                to_native(reboot_result['stdout'].strip()), to_native(reboot_result['stderr'].strip()))
+            result['msg'] = "Reboot command failed. Error was {stdout}, {stderr}".format(
+                stdout=to_native(reboot_result['stdout'].strip()),
+                stderr=to_native(reboot_result['stderr'].strip()))
             return result
 
         result['failed'] = False
 
         # attempt to store the original connection_timeout option var so it can be reset after
-        self._original_connection_timeout = None
         try:
             self._original_connection_timeout = self._connection.get_option('connection_timeout')
         except AnsibleError:
-            display.debug("%s: connect_timeout connection option has not been set" % self._task.action)
+            display.debug("{action}: connect_timeout connection option has not been set".format(action=self._task.action))
 
         return result
 
-    def validate_reboot(self):
-        display.debug('%s: Validating reboot' % self._task.action)
+    def validate_reboot(self, distribution, action_kwargs=None):
+        display.vvv('{action}: validating reboot'.format(action=self._task.action))
         result = {}
 
         try:
             # keep on checking system boot_time with short connection responses
             reboot_timeout = int(self._task.args.get('reboot_timeout', self._task.args.get('reboot_timeout_sec', self.DEFAULT_REBOOT_TIMEOUT)))
             connect_timeout = self._task.args.get('connect_timeout', self._task.args.get('connect_timeout_sec', self.DEFAULT_CONNECT_TIMEOUT))
-            self.do_until_success_or_timeout(self.check_boot_time, reboot_timeout, action_desc="boot_time check")
+
+            # if action_kwargs is None:
+            #     action_kwargs = {}
+            self.do_until_success_or_timeout(
+                action=self.check_boot_time,
+                action_desc="boot_time check",
+                reboot_timeout=reboot_timeout,
+                distribution=distribution,
+                action_kwargs=action_kwargs)
 
             if connect_timeout:
                 try:
                     self._connection.set_option("connection_timeout", connect_timeout)
```

Comment:
I know this isn't part of the changes but I just found this. `connect_timeout` is the value of `connect_timeout` or `connect_timeout_sec` as part of the reboot task. I believe we want to use `self._original_connection_timeout` if it has been set instead.

Reply:
Good catch. I will fix this.